src/main/java/edu/cornell/kfs/fp/batch/service/impl/CuBudgetAdjustmentDocumentGenerator.java

```diff
@@ -0,0 +1,96 @@
+        BudgetAdjustmentAccountingLine baLine = (BudgetAdjustmentAccountingLine) accountingLine;
+        
+        setAmountIfNotNull(baLine::setBaseBudgetAdjustmentAmount, xmlLine.getBaseAmount());
+        setAmountIfNotNull(baLine::setCurrentBudgetAdjustmentAmount, xmlLine.getAmount());
+        setAmountIfNotNull(baLine::setFinancialDocumentMonth1LineAmount, xmlLine.getMonth01Amount());
+        setAmountIfNotNull(baLine::setFinancialDocumentMonth2LineAmount, xmlLine.getMonth02Amount());
+        setAmountIfNotNull(baLine::setFinancialDocumentMonth3LineAmount, xmlLine.getMonth03Amount());
+        setAmountIfNotNull(baLine::setFinancialDocumentMonth4LineAmount, xmlLine.getMonth04Amount());
+        setAmountIfNotNull(baLine::setFinancialDocumentMonth5LineAmount, xmlLine.getMonth05Amount());
+        setAmountIfNotNull(baLine::setFinancialDocumentMonth6LineAmount, xmlLine.getMonth06Amount());
+        setAmountIfNotNull(baLine::setFinancialDocumentMonth7LineAmount, xmlLine.getMonth07Amount());
+        setAmountIfNotNull(baLine::setFinancialDocumentMonth8LineAmount, xmlLine.getMonth08Amount());
+        setAmountIfNotNull(baLine::setFinancialDocumentMonth9LineAmount, xmlLine.getMonth09Amount());
+        setAmountIfNotNull(baLine::setFinancialDocumentMonth10LineAmount, xmlLine.getMonth10Amount());
+        setAmountIfNotNull(baLine::setFinancialDocumentMonth11LineAmount, xmlLine.getMonth11Amount());
+        setAmountIfNotNull(baLine::setFinancialDocumentMonth12LineAmount, xmlLine.getMonth12Amount());
+        
+        return accountingLine;
+    }
+
+    protected <N extends Number> void setAmountIfNotNull(Consumer<N> amountSetter, N amount) {
+        if (ObjectUtils.isNotNull(amount)) {
+            amountSetter.accept(amount);
+        }
+    }
+
+    @Override
+    protected void populateCustomAccountingDocumentData(
+            CuBudgetAdjustmentDocument document, AccountingXmlDocumentEntry documentEntry) {
+        super.populateCustomAccountingDocumentData(document, documentEntry);
+        validateAndSetFiscalYear(document, documentEntry);
+    }
+
+    protected void validateAndSetFiscalYear(CuBudgetAdjustmentDocument document, AccountingXmlDocumentEntry documentEntry) {
+        Integer fiscalYear = documentEntry.getPostingFiscalYear();
+        if (ObjectUtils.isNull(fiscalYear)) {
+            throw new RuntimeException("Fiscal year cannot be null");
+        }
+        
+        @SuppressWarnings("unchecked")
+        List<FiscalYearFunctionControl> allowedYears = fiscalYearFunctionControlService.getBudgetAdjustmentAllowedYears();
+        boolean fiscalYearAllowed = allowedYears.stream()
+                .anyMatch((fyFunctionControl) -> fiscalYear.equals(fyFunctionControl.getUniversityFiscalYear()));
+        if (!fiscalYearAllowed) {
+            throw new RuntimeException("Budget Adjustments are not allowed for fiscal year: " + fiscalYear);
```

Comment:
Should this be a ValidationException, so it can be handled by the report correctly?

Reply:
Ok, I will update the exception type accordingly.